Containers/talk/start.sh

```diff
@@ -16,43 +16,46 @@ elif [ -z "$INTERNAL_SECRET" ]; then
 fi
 
 set -x
-IPv4_ADDRESS_TALK="$(dig nextcloud-aio-talk A +short)"
+if [ -n "$(dig nextcloud-aio-talk A +short | grep -P "^[0-9.]+$" | head -n1)" ]; then
+    IPv4_ADDRESS_TALK="$(dig nextcloud-aio-talk A +short | grep -P "^[0-9.]+$" | head -n1)"
+fi
+if [ -n "$(dig nextcloud-aio-talk AAAA +short | grep -P "^[0-9a-fA-F:]+$" | head -n1)" ]; then
+    IPv6_ADDRESS_TALK="$(dig nextcloud-aio-talk AAAA +short | grep -P "^[0-9a-fA-F:]+$" | head -n1)"
+fi
+
+if [ -n "$(dig "$NC_DOMAIN" A +short | grep -P "^[0-9.]+$" | head -n1)" ]; then
+    IPv4_ADDRESS_NC="$(dig "$NC_DOMAIN" A +short | grep -P "^[0-9.]+$" | head -n1)"
+fi
+if [ -n "$(dig "$NC_DOMAIN" AAAA +short | grep -P "^[0-9a-fA-F:]+$" | head -n1)" ]; then
+    IPv6_ADDRESS_NC="$(dig "$NC_DOMAIN" AAAA +short | grep -P "^[0-9a-fA-F:]+$" | head -n1)"
+fi
 set +x
 
 # Turn
-cat << TURN_CONF > "/etc/turnserver.conf"
-listening-port=$TALK_PORT
-fingerprint
-use-auth-secret
-static-auth-secret=$TURN_SECRET
-realm=$NC_DOMAIN
-total-quota=0
-bps-capacity=0
-stale-nonce
-no-multicast-peers
-simple-log
-pidfile=/var/tmp/turnserver.pid
-no-tls
-no-dtls
-userdb=/var/lib/turn/turndb
-# Based on https://nextcloud-talk.readthedocs.io/en/latest/TURN/#turn-server-and-internal-networks
-allowed-peer-ip=$IPv4_ADDRESS_TALK
-denied-peer-ip=0.0.0.0-0.255.255.255
-denied-peer-ip=10.0.0.0-10.255.255.255
-denied-peer-ip=100.64.0.0-100.127.255.255
-denied-peer-ip=127.0.0.0-127.255.255.255
-denied-peer-ip=169.254.0.0-169.254.255.255
-denied-peer-ip=172.16.0.0-172.31.255.255
-denied-peer-ip=192.0.0.0-192.0.0.255
-denied-peer-ip=192.0.2.0-192.0.2.255
-denied-peer-ip=192.88.99.0-192.88.99.255
-denied-peer-ip=192.168.0.0-192.168.255.255
-denied-peer-ip=198.18.0.0-198.19.255.255
-denied-peer-ip=198.51.100.0-198.51.100.255
-denied-peer-ip=203.0.113.0-203.0.113.255
-denied-peer-ip=240.0.0.0-255.255.255.255
+cat << TURN_CONF > "/etc/eturnal.yml"
+eturnal:
+  listen:
+    - ip: "::"
```

Comment:
This looks like ipv6-syntax. Does it still listen on ipv4-addresses?

Reply:
yes